### PR TITLE
fetch aws config details from terraform

### PR DESCRIPTION
**Pull request by Sahiba3108**
### :nut_and_bolt: Description: What code changed, and why?
For AWS deployment, the config file doesn’t have the node's IP address. 
So the config validator should fetch the required details before validating the config and pass the required config to batch check API.

### :chains: Related Resources
https://chefio.atlassian.net/browse/CHEF-3505
### :+1: Definition of Done
Fetch the Aws config Details like Private ips of Automate, Chef server, Pg, Os, Automate LB fqdn, Bucket name if we have used s3 as backup config from terraform file.
### :athletic_shoe: How to Build and Test the Change
Invoke this fetchAwsConfigFromTerraform() method, it will return the config fetched from terraform file i.e. output.auto.tfvars
### :white_check_mark: Checklist

**All PRs** must tick these:

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

With occasional exceptions, all PRs **from Progress employees** must tick these:

- [x] Is the code clear? *(complicated code or lots of comments--subdivide and use well-named methods, meaningful variable names, etc.)*
- [x] Consistency checked? *(user notifications, user prompts, visual patterns, code patterns, variable names)*
- [x] Repeated code blocks eliminated? *(adapt and reuse existing components, blocks, functions, etc.)*
- [x] Spelling, grammar, typos checked? *(at a minimum use `make spell` in any component directory)*
- [x] Code well-formatted? *(indents, line breaks, etc. improve rather than hinder readability)*

All PRs **from Progress employees** should tick these if appropriate:

- [x] Tests added/updated? (all new code needs new tests)
- [ ] Docs added/updated? (all customer-facing changes)

*Please add a note next to any checkbox above if you are NOT ticking it.*

### :camera: Screenshots, if applicable
Screenshot 2023-06-20 at 6 20 40 PM

### CODE REVIEW

File: components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+		return nil, err
+	}
+	return awsAutoTfvarConfig, nil
+}
+
+func getJsonFromTerraformOutputAutoTfVarsFile(jsonString string) (*configDetails, error) {
+	params := configDetails{}
+	err := json.Unmarshal([]byte(jsonString), &params)
+	if err != nil {
+		writer.Fail(err.Error())
```
Comment (iamazzeez) [inline components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails.go:34]: writer.Fail(err.Error()) is this needed here?,  also we can json.Unmarshal([]byte(jsonString), &configDetails{}) and simply return configDetails

File: components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	err := json.Unmarshal([]byte(AwsAutoTfvarsJsonString), &params)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	dirPath := filepath.Join(initConfigHabA2HAPathFlag.a2haDirPath, "terraform", "reference_architectures", "deployment")
+	filePath := dirPath + "/output.auto.tfvars"
+	err = os.MkdirAll(dirPath, os.ModePerm)
+	if err != nil {
```
Comment (iamazzeez) [inline components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails_test.go:40]: we can simply use assert.NoError(t, err, "Error creating directories") at all errors like in line 80

File: components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails_test.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+			ExpectedError:  errors.New(""),
+		},
+	}
+
+	for _, e := range tests {
+		_, err = file.WriteString(e.WriteString)
+		assert.NoError(t, err)
+		t.Run(e.TestName, func(t *testing.T) {
+			res, err := fetchAwsConfigFromTerraform()
+			if e.ExpectedError != nil {
```
Comment (iamazzeez) [inline components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails_test.go:77]: Do we need if and else here? 

assert.NoError(t, err)
assert.Equal(t, e.ExpectedOutput, res) only this should work.

File: components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails_test.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, e.ExpectedOutput, res)
+		})
+	}
+
+	err = file.Close()
+	if err != nil {
```
Comment (iamazzeez) [inline components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails_test.go:87]: assert.NoError(t, err)

File: components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails_test.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+		_, err = file.WriteString(e.WriteString)
+		assert.NoError(t, err)
+		t.Run(e.TestName, func(t *testing.T) {
+			res, err := fetchAwsConfigFromTerraform()
+			if e.ExpectedError != nil {
+				assert.Error(t, err)
+			} else {
+				assert.NoError(t, err)
+			}
+			assert.Equal(t, e.ExpectedOutput, res)
```
Comment (iamazzeez) [inline components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails_test.go:82]: Do we need if and else here?

assert.NoError(t, err)
assert.Equal(t, e.ExpectedOutput, res) only this should work.

Comment (Sahiba3108) [reply components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails_test.go:82]: There are some scenarios where we can expect the error also, so i have handled it. If i remove the if and else condition then those test case will fail.

File: components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails_test.go (hunk 2 above)
Comment (Sahiba3108) [reply components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails_test.go:40]: done

File: components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails.go (hunk 1 above)
Comment (Sahiba3108) [reply components/automate-cli/cmd/chef-automate/fetchAwsConfigDetails.go]: removed writer.Fail(err.Error())

**Review by Dmaddu — APPROVED** [review]

**Pull request merged**